src/package-managers/bun.ts

```diff
@@ -0,0 +1,42 @@
+import spawn from 'spawn-please'
+import { Index } from '../types/IndexType'
+import { NpmOptions } from '../types/NpmOptions'
+import { latest as npmLatest, list as npmList } from './npm'
+
+/**
+ * (Bun) Fetches the list of all installed packages.
+ */
+export const list = npmList
```

Comment:
You should spawn bun for the list command, as npm may have a different global install directory than bun.

Hopefully bun has a `--json` option, otherwise you'll have to parse the results.